crates/bevy_derive/src/lib.rs

```diff
@@ -19,8 +19,15 @@ use proc_macro::TokenStream;
 use quote::format_ident;
 
 /// Generates a dynamic plugin entry point function for the given `Plugin` type.
+///
+/// This is deprecated since 0.14. The current dynamic plugin system is unsound and will be removed in 0.15.
```

Comment:
Just as a note, the `deprecated` attribute for proc macros does not work, which is why I'm also including the comment.